sklearn/manifold/t_sne.py

```diff
@@ -455,18 +455,24 @@ def trustworthiness(X, X_embedded, n_neighbors=5,
                       "parameter instead.", DeprecationWarning)
         metric = 'precomputed'
     dist_X = pairwise_distances(X, metric=metric)
+    dist_X += np.diag(np.full(dist_X.shape[0], np.inf))
```

Comment:
When `metric='precomputed'`, we have `dist_X is X`, so we should not modify `X` inplace.
You can add something like:
```py
if metric == 'precomputed':
    dist_X = dist_X.copy()
```

Reply:
Oh that's right sorry, I didn't know `+=` was in fact inplace with numpy
I updated the code with the copy and using `fill_diagonal` (which I can use now since it's a copy)